src/framework/route/core.clj

```diff
@@ -38,7 +38,7 @@
           (?assoc-in [:request-data :match] match)
           (?assoc-in [:request-data :permission] permission)
           (assoc-in [:request-data :action]
-                    (or (if (jetty/ws-upgrade-request? request)
+                    (or (if (and (jetty/ws-upgrade-request? request) ws-action)
```

Comment:
It seems it is this way to avoid returning nil on line 42. LGTM

Reply:
> It seems it is this way to avoid returning nil on line 42. LGTM

I've pushed a simplified action resolver